### ISSUE TITLE
Optimize node builds to cache node_modules

### ISSUE THREAD

**Issue by mtheoryx** [MichaelJolley/io#45]
```
WORKDIR /app

COPY ./src/admin .
COPY ./src/common ./src

WORKDIR /app/src

RUN npm install

WORKDIR /app

RUN npm install
```

This is going to end up busting the layer cache whenever application code changes. The truth is, npm dependencies change far less often than code changes.

[The proper order is similar to this example](https://github.com/mtheoryx/bottibles/blob/master/chat-reader/Dockerfile)

Important to note:

1. Copy your package.json and do npm install,
2. THEN copy in your app code

This allows faster builds and you don't need to build new layers of npm dependencies when only your code changes.

**Comment by MichaelJolley** [comment]
I'm glad you brought this up.  I've seen you talk about this but haven't really understood it.  Almost started working on it on stream today but wanted to get your feedback before tackling it.

**Comment by mtheoryx** [comment]
Fixed by #55 !!

![bitmoji](https://render.bitstrips.com/v2/cpanel/c08ae70b-d764-4f67-a249-6f2fe1b8a9c4-c3b45c6e-2943-4fd5-ab6e-fe23bffcb90b-v1.png?transparent=1&palette=1&width=246)
